src/renderer/components/ReaderApp.tsx

```diff
@@ -289,36 +528,36 @@ export default class ReaderApp extends React.Component<undefined, ReaderAppState
             // }
 
             if (title) {
-                console.log(title);
+                // console.log(title);
                 window.document.title = "Readium2 [ " + title + "]";
                 this.setState({
                     title,
                 });
             }
         }
 
-        if (_publication.Spine && _publication.Spine.length) {
-            console.log(_publication.Spine);
+        if (publication.Spine && publication.Spine.length) {
+            // console.log(publication.Spine);
             const links: IStringMap = {};
-            _publication.Spine.forEach((spineItemLink) => {
-                links[spineItemLink.Href] = publicationJsonUrl + "/../" + spineItemLink.Href;
+            publication.TOC.forEach((spineItemLink) => {
+                links[spineItemLink.Title] = publicationJsonUrl + "/../" + spineItemLink.Href;
```

Comment:
`spineItemLink.Title` is not guaranteed to be unique (and in some cases is undefined / null), therefore I would suggest to use the `spineItemLink.Href` key instead.